braindecode/models/sccnet.py

```diff
@@ -0,0 +1,176 @@
+"""Spatial component-wise convolutional network (SCCNet) for motor-imagery EEG
+    classification.
+Authors: Chun-Shu Wei
+         Bruno Aristimunha <b.aristimunha@gmail.com> (braindecode adaptation)
+
+Wei, C. S., Koike-Akino, T., & Wang, Y. (2019, March). Spatial component-wise
+convolutional network (SCCNet) for motor-imagery EEG classification. In 2019
+9th International IEEE/EMBS Conference on Neural Engineering (NER) (pp. 328-331).
+IEEE.
+"""
+
+import math
+import torch
+from torch import nn
+
+from einops.layers.torch import Rearrange
+from braindecode.models.base import EEGModuleMixin
+from braindecode.models.modules import LogActivation
+
+
+class SCCNet(EEGModuleMixin, nn.Module):
+    """SCCNet from Wei, C S (2019) [sccnet]_.
+
+    Spatial component-wise convolutional network (SCCNet) for motor-imagery EEG
+    classification.
+
+    .. figure:: https://dt5vp8kor0orz.cloudfront.net/6e3ec5d729cd51fe8acc5a978db27d02a5df9e05/2-Figure1-1.png
+       :align: center
+       :alt:  Spatial component-wise convolutional network
+
+
+    1. **Spatial Component Analysis**: Performs convolution spatial filtering
+        across all EEG channels to extract spatial components, effectively
+        reducing the channel dimension.
+    2. **Spatio-Temporal Filtering**: Applies convolution across the spatial
+        components and temporal domain to capture spatio-temporal patterns.
+    3. **Temporal Smoothing (Pooling)**: Uses average pooling over time to smooth the
+       features and reduce the temporal dimension, focusing on longer-term patterns.
+    4. **Classification**: Flattens the features and applies a fully connected
+       layer.
+
+
+    Parameters
+    ----------
+    n_filters_spat : int, optional
+        Number of spatial filters in the first convolutional layer. Default is 22.
+    n_filters_spat_filt : int, optional
```

Comment:
```suggestion
    n_spatial_filters2 : int, optional
```
or `n_spatial_filters_smooth`